app/assets/javascripts/maps.js

```diff
@@ -0,0 +1,61 @@
+//= require google-infobox/google-infobox
+//= require google_maps/custom_marker
+
+var map;
+var Settings = {
+  id: 'map-canvas',
+  lat: 51.5978,
+  lng: -0.3370,
+  zoom: 12
+};
+
+function initMap() {
+
+  map = new google.maps.Map(document.getElementById(Settings.id), {
+    center: {lat: Settings.lat, lng: Settings.lng},
+    zoom: Settings.zoom
+  });
+
+  var markerData = $("#marker_data").data().markers;
+
+  var marker;
+
+  var ib = new InfoBox();
+  var ibOptions = {
+    pixelOffset: new google.maps.Size(-151, 10)
+  };
+
+  var boxText = document.createElement("div");
+  boxText.setAttribute("class", "arrow_box")
+  ibOptions.content = boxText;
+
+  markerData.forEach(function(item) {  
+    var latLng = new google.maps.LatLng(item.lat, item.lng);
+
+    marker = new CustomMarker(
+        latLng,
+        map,
+        {
+          content: item.custom_marker
+        }
+        );
+
+    google.maps.event.addListener(marker, 'click', (function(marker, item) {
+
+      return function() {
+
+        ib.setOptions(ibOptions);
+        boxText.innerHTML = item.infowindow;
+        $(ib.content_).find('.close').click(function(){
+          ib.close();
+        });
+        ib.open(map, marker);
+        map.panTo(ib.getPosition());
+
+      }
+
+    })(marker, item)); 
```

Comment:
Code climate [noticed](https://codeclimate.com/github/AgileVentures/LocalSupport/pull/397) that this function is invoked when passed to the `addListener` function. It's interesting that this invoked function returns a function. @Maroo-b, does `marker` and `item` still work in the inner-most function without being enclosed in this invoked function?